### PR TITLE
feat: Prediction with more coins

### PR DESCRIPTION
**Pull request by ChefMomota**
### Summary
🚀📦🏷️

This pull request adds multi-chain support to the prediction feature, refactors the prediction-related types and constants to use a shared package, and fixes some TypeScript and React errors and warnings. It modifies several files in the `apps/web` directory, including the menu config, the subgraph health indicators, the prediction page, the prediction state, and the prediction components. It also adds a new dependency to the `apps/web/package.json` file.

> _The prediction feature got a new package_
> _With types and constants to manage_
> _They updated imports and props_
> _And fixed some errors and ops_
> _To make it work with different chains of cabbage_

### Walkthrough
*  Add a new dependency for the prediction package and update the imports of the types and constants that are moved to the package ([link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8R61), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-a1ed63a2f65f726a02523d0f9389df53bd35324d627439d2c1ce0ad2a453883fL1-R1), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-e0daeed87867fc4fc6b0b1a9c2d52c9bfbd175cd1822e100f43300c4046484adR5), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-a8f99e5f4e69d35778497b7fe5b2dfa26da8c67a2731b50f0bde548c80b4ceefL6-R7), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-d26889446d6fdd4d786ce8e629ede14174c310bd25ce2bb423b5d46db82de68dR1), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-a10dd099b9e7b9f90a9c47622f2190bed7f23cf869ae94058dcfe78abcd4884fL2-R5), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-a10dd099b9e7b9f90a9c47622f2190bed7f23cf869ae94058dcfe78abcd4884fR12-R17), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-49bff92257ec4777477d6fd1bb9f82d8a47549f988d572d51ddff77dec6e705cR6), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-f5c35970b522c843c38eb58b8ec087c69542a94e54092f8807eb27a82ac69711L11-R16), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1e18a9b62e159a571e35f91dcd049d12d671d398ae039cbb6208cf82a023c9e6R3), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-aa81f44b73af3bb717f598cda55216c1a4531c87248b2a5782fa98b936e9ce12L2-R3), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-e60d214e0b007d637e1805d6cad32ab086d2c476d759fa151caf9ba674d9e375R5), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-413567f1660e2ebe00f10e81aa88b7a26f80cf7f8d4aecd0b38efba588992ca2R5), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL15-R15), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-caab29b952970a40e76d80a56aa8e4563bc394c449d15d2f3bd5769acd318b40L3-R3), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-3b422f15893d9a832f054eb8664ed80b6015e5c32975e55fae485909c4c2c7f2L5-R5), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-e534ec68945e2c11520256c3dcbefb8c51337966af3dffb29e6312a64ad4154eL7-R7))
*  Add a new property to the prediction menu item and the prediction page component to specify the supported chain IDs for the prediction feature ([link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-e0daeed87867fc4fc6b0b1a9c2d52c9bfbd175cd1822e100f43300c4046484adR153), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-d26889446d6fdd4d786ce8e629ede14174c310bd25ce2bb423b5d46db82de68dR10))
*  Remove the prediction feature from the text that informs the user about the BNB-only features ([link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-dada85db7cfee99b5258db2b6974b0fcb543d6cb2d77e5b60e7e2657d8acf005L55-R55))
*  Add null checks and optional chaining operators to prevent potential errors when accessing the prediction subgraph data or the element references ([link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-a8f99e5f4e69d35778497b7fe5b2dfa26da8c67a2731b50f0bde548c80b4ceefL20-R25), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-a8f99e5f4e69d35778497b7fe5b2dfa26da8c67a2731b50f0bde548c80b4ceefL29-R33), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1e18a9b62e159a571e35f91dcd049d12d671d398ae039cbb6208cf82a023c9e6L36-R64), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1e18a9b62e159a571e35f91dcd049d12d671d398ae039cbb6208cf82a023c9e6L77-R86), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1e18a9b62e159a571e35f91dcd049d12d671d398ae039cbb6208cf82a023c9e6L84-R93), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL117-R119), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL126-R139), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL143-R155), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL161-R177))
*  Add a chainId parameter to the functions that query the prediction subgraphs and use the appropriate subgraph URL for the given chainId ([link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-a10dd099b9e7b9f90a9c47622f2190bed7f23cf869ae94058dcfe78abcd4884fL99-R105), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-a10dd099b9e7b9f90a9c47622f2190bed7f23cf869ae94058dcfe78abcd4884fL113-R116), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-413567f1660e2ebe00f10e81aa88b7a26f80cf7f8d4aecd0b38efba588992ca2R34), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-413567f1660e2ebe00f10e81aa88b7a26f80cf7f8d4aecd0b38efba588992ca2L38-R55))
*  Add a type annotation of any to the preloadedState parameter of the makeStore function to fix a TypeScript error ([link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-8a3ad58e4950a8d1d5f4dab5c642ebd20346ba9590cad3beb70134fcf7f9166bL3-R3))
*  Change the type of the account parameter from string to Address in the prediction hooks and selectors to match the type used in the prediction package ([link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-49bff92257ec4777477d6fd1bb9f82d8a47549f988d572d51ddff77dec6e705cL31-R32), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1e18a9b62e159a571e35f91dcd049d12d671d398ae039cbb6208cf82a023c9e6L13-R16))
*  Remove the declaration and export of the types that are moved to the prediction package from the state/types file ([link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-f5c35970b522c843c38eb58b8ec087c69542a94e54092f8807eb27a82ac69711L81-R123), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-e60d214e0b007d637e1805d6cad32ab086d2c476d759fa151caf9ba674d9e375L179-L201))
*  Use the $ prefix for the props that are used for styling only and not passed to the underlying elements in the styled components (  [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL60-R66), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL91-R93), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL205-R215), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL273-R277))
*  Memoize the token symbol value using the useMemo hook and use it instead of the token.symbol value that may be undefined ([link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL1-R1), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL126-R139), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL246-R250), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL252-R256), [link](https://github.com/pancakeswap/pancake-frontend/pull/8417/files?diff=unified&w=0#diff-1ba30f46c43bce11a182e5ad473180b3e1dd632de0a565fce351b59c8f2a744bL263-R267))

---

## PR-Codex overview
This PR focuses on updating the code related to the Predictions feature. 

### Detailed summary
- Updated predictionV1 and predictionV2 ABIs.
- Deleted prediction index files.
- Exported getPredictionConfig function.
- Updated supportedChains and config constants.
- Updated handleConfirm function.
- Updated BunnySlider component.
- Added predictions record.
- Updated getTimePeriods file.
- Added vitest.config.ts file.
- Updated cakePredictions ABI.
- Updated handleEvent function.
- Updated imports in test file.
- Updated imports in various files.
- Added prediction package to package.json.
- Updated useStableSwapInfo file.
- Updated MobileMenu file.
- Updated helpers file.
- Added useGetNativeTokenBalance function.
- Updated GRAPH_API_PREDICTION_BNB endpoint.

> The following files were skipped due to too many changes: `apps/web/src/config/constants/endpoints.ts`, `packages/prediction/package.json`, `apps/web/src/views/Predictions/Leaderboard/components/MobileBetsTable.tsx`, `apps/web/src/views/Predictions/hooks/usePredictionConfigs.ts`, `apps/web/src/views/Predictions/hooks/usePredictionToken.ts`, `apps/web/src/components/Menu/config/config.ts`, `apps/web/src/views/Predictions/context/SwiperProvider.tsx`, `apps/web/src/views/Predictions/components/ReclaimPositionButton.tsx`, `apps/web/src/state/multicall/hooks.ts`, `packages/prediction/src/constants/config.ts`, `apps/web/src/views/Predictions/components/CollectWinningsButton.tsx`, `apps/web/src/views/Predictions/Leaderboard/components/Results/MobileResults.tsx`, `apps/web/src/pages/prediction/leaderboard.tsx`, `packages/prediction/src/constants/predictions/zkSync.ts`, `apps/web/src/hooks/useBNBPrice.ts`, `apps/web/src/hooks/useCakePrice.ts`, `apps/web/src/views/Predictions/Positions.tsx`, `packages/prediction/src/chainlinkOracleContract.ts`, `packages/prediction/src/endpoints.ts`, `packages/prediction/src/predictionContract.ts`, `apps/web/src/views/Predictions/components/RoundCard/MultiplierArrow.tsx`, `apps/web/src/views/Predictions/hooks/useGaletoOraclePrice.ts`, `apps/web/src/views/Predictions/components/CardFlip.tsx`, `apps/web/src/views/Predictions/Leaderboard/components/Results/DesktopRow.tsx`, `apps/web/src/state/predictions/bnbQueries.ts`, `apps/web/src/components/TokenImage/index.tsx`, `apps/web/src/state/predictions/cakeQueries.ts`, `apps/web/src/views/Predictions/components/CollectWinningsPopup.tsx`, `apps/web/src/state/types.ts`, `apps/web/src/views/Predictions/components/RoundCard/SoonRoundCard.tsx`, `apps/web/src/views/Predictions/components/TokenSelector/Price.tsx`, `apps/web/src/hooks/useBUSDPrice.ts`, `apps/web/src/__tests__/state/predictions/helpers.test.ts`, `packages/prediction/src/constants/predictions/bsc.ts`, `apps/web/src/views/Predictions/hooks/usePollPredictions.ts`, `apps/web/src/config/constants/contracts.ts`, `apps/web/src/views/Predictions/History.tsx`, `packages/prediction/src/type.ts`, `apps/web/src/state/predictions/newTokenQueries.ts`, `apps/web/src/hooks/useContract.ts`, `apps/web/src/views/Predictions/components/Notification/Notification.tsx`, `apps/web/src/state/predictions/newTransformers.ts`, `apps/web/src/views/Predictions/index.tsx`, `apps/web/src/views/Predictions/Leaderboard/components/Results/MobileRow.tsx`, `apps/web/src/views/Predictions/hooks/useIsRefundable.ts`, `apps/web/src/views/Predictions/Leaderboard/components/Results/DesktopResults.tsx`, `apps/web/src/views/Predictions/hooks/usePollOraclePrice.ts`, `apps/web/src/state/predictions/queries.ts`, `apps/web/src/views/Predictions/components/RoundResult/RoundResultHistory.tsx`, `apps/web/src/views/Predictions/Leaderboard/components/AddressSearch.tsx`, `apps/web/src/views/Predictions/hooks/useTokenPrice.ts`, `apps/web/src/views/Predictions/Leaderboard/components/WalletStatsModal.tsx`, `apps/web/src/views/Predictions/components/TokenSelector/Desktop.tsx`, `apps/web/src/views/Predictions/MobileChart.tsx`, `apps/web/src/views/Predictions/Leaderboard/components/Results/styles.tsx`, `apps/web/src/views/Predictions/components/Menu.tsx`, `apps/web/src/views/Predictions/components/RoundCard/ExpiredRoundCard.tsx`, `apps/web/src/views/Predictions/Mobile.tsx`, `apps/web/src/components/SubgraphHealthIndicator/FloatingSubgraphHealthIndicators.tsx`, `apps/web/src/views/Predictions/components/RoundCard/EnteredTag.tsx`, `apps/web/src/views/Predictions/components/TokenSelector/Mobile.tsx`, `packages/uikit/src/components/Select/Select.tsx`, `apps/web/src/views/Predictions/components/v1/V1ClaimCheck.tsx`, `apps/web/src/utils/contractHelpers.ts`, `apps/web/src/views/Predictions/Leaderboard/components/Results/ConnectedWalletResult.tsx`, `apps/web/src/views/Predictions/components/History/RoundsTab.tsx`, `apps/web/src/views/Predictions/Leaderboard/components/PreviousBetsTable.tsx`, `apps/web/src/views/Predictions/components/RoundCard/index.tsx`, `apps/web/src/views/Predictions/Leaderboard/components/Results/index.tsx`, `apps/web/src/views/Predictions/components/RoundCard/OpenRoundCard.tsx`, `apps/web/src/views/Predictions/Leaderboard/components/Results/ResultAvatar.tsx`, `apps/web/src/state/predictions/selectors.ts`, `apps/web/src/state/predictions/tokenTransformers.ts`, `packages/localization/src/config/translations.json`, `apps/web/src/views/Predictions/context/PredictionConfigProviders.tsx`, `apps/web/src/views/Predictions/Leaderboard/index.tsx`, `apps/web/src/views/Predictions/Leaderboard/components/Results/RankingCard.tsx`, `apps/web/src/state/predictions/hooks.ts`, `apps/web/src/views/Predictions/components/TokenSelector/SvgToken.tsx`, `apps/web/src/views/Predictions/components/CollectRoundWinningsModal.tsx`, `apps/web/src/views/Predictions/components/Label.tsx`, `apps/web/src/views/Predictions/components/History/HistoricalBet.tsx`, `apps/web/src/views/Predictions/components/TokenSelector/index.tsx`, `pnpm-lock.yaml`, `apps/web/src/views/Predictions/components/RoundCard/LiveRoundCard.tsx`, `apps/web/src/views/Predictions/components/RoundResult/styles.tsx`, `apps/web/src/views/Predictions/components/History/PnlTab/PnlTab.tsx`, `packages/uikit/src/components/Slider/styles.ts`, `apps/web/src/views/Predictions/components/ChainlinkChart.tsx`, `apps/web/src/views/Predictions/components/History/BetResult.tsx`, `apps/web/src/views/Predictions/Leaderboard/components/Filters/index.tsx`, `apps/web/src/views/Predictions/components/RoundCard/SetPositionCard.tsx`, `apps/web/src/state/predictions/helpers.ts`, `apps/web/src/views/Predictions/Desktop.tsx`, `apps/web/src/state/predictions/index.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/hooks/useStableSwapInfo.ts [hunk 1]
```diff
@@ -53,14 +53,14 @@ export function useStableSwapInfo(stableSwapAddress: Address | undefined, lpAddr
     ],
   })
 
-  const feeNumerator = results?.[4]?.result
-  const feeDenominator = results?.[5]?.result
+  const feeNumerator = Number(results?.[4]?.result)
```
Comment (chef-jojo) [inline apps/web/src/hooks/useStableSwapInfo.ts:56]: avoid casting bigint into number here

File: apps/web/src/hooks/useStableSwapInfo.ts [hunk 2]
```diff
@@ -53,14 +53,14 @@ export function useStableSwapInfo(stableSwapAddress: Address | undefined, lpAddr
...
+  const feeDenominator = Number(results?.[5]?.result)
 
   return {
     balances: [results?.[0].result, results?.[1].result],
     amplifier: results?.[2].result,
     totalSupply: results?.[3].result,
-    fee: feeNumerator && feeDenominator && new Percent(feeNumerator, feeDenominator),
+    fee: new Percent(feeNumerator, feeDenominator),
```
Comment (chef-jojo) [inline apps/web/src/hooks/useStableSwapInfo.ts:63]: this will throw error if it's not a valid number

File: packages/prediction/package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "type": "module",
+  "peerDependencies": {
+    "@pancakeswap/sdk": "workspace:*"
+  },
+  "dependencies": {
+    "@pancakeswap/chains": "workspace:*",
+    "@pancakeswap/tokens": "workspace:*",
+    "@pancakeswap/pools": "workspace:*",
+    "@pythnetwork/pyth-evm-js": "^1.29.2",
+    "@tanstack/react-query": "^4.28.0",
```
Comment (chefjackson) [inline packages/prediction/package.json:18]: the sdk shouldn't be relying on usequery. Better practice would be sdk handling the request logic and let the web application to deal with hooks

Comment (ChefMomota) [reply packages/prediction/package.json]: done

**Pull request merged**